go/geth_adapter/adapter.go

```diff
@@ -480,7 +481,22 @@ func (a *runContextAdapter) IsSlotInAccessList(addr vm.Address, key vm.Key) (add
 }
 
 func (a *runContextAdapter) HasSelfDestructed(addr vm.Address) bool {
-	return a.evm.StateDB.HasSuicided(gc.Address(addr))
+	return a.evm.StateDB.HasSelfDestructed(gc.Address(addr))
+}
+
+func uint256ToValue(value *uint256.Int) (result vm.Value, err error) {
+	if value == nil {
+		return result, fmt.Errorf("unable to convert nil to Hash")
+	}
+	if value.ByteLen() > 32 {
+		return result, fmt.Errorf("value exceeds maximum value for Hash, %v of 32 bytes max", len(value.Bytes()))
+	}
+	result = value.Bytes32()
+	return result, nil
+}
+
+func valueToUint256(value vm.Value) *uint256.Int {
+	return uint256.NewInt(0).SetBytes(value[:])
```

Comment:
This might be future candidates for public utility functions in the `vm` package.

Reply:
I think this should be done in a separate PR to not move unrelated code in this PR.